nodes/ElseNode.java

```diff
@@ -57,24 +91,29 @@ static public ElseNode parseElseNode(ArrayList<Token> tokens) {
 
             // if no left brace {, throw exception
             if (tokens.get(0).getTokenType() != TokenType.L_BRACE) {
-                throw new SyntaxException(tokens.get(0).getToken(), tokens.get(0).getFilename(), tokens.get(0).getLineNum());
+                throw new SyntaxException("Else statement missing left brace", tokens.get(0).getToken(), tokens.get(0).getFilename(), tokens.get(0).getLineNum());
             }
 
             // remove left brace {
             tokens.remove(0);
 
+            Token fileInfo = null;
+            if (!(tokens.isEmpty())) {
+                fileInfo = tokens.get(0);
```

Comment:
good idea, I'll probably do something similar to make semantic error msgs better